### ISSUE TITLE
Panic when decoding Avro: range end index 10 out of range for slice of length 0

### ISSUE THREAD

**Issue by mlazowik** [MaterializeInc/materialize#7105]
### What version of Materialize are you using?

```
0.8.0
```

### How did you install Materialize?

* [x] Docker image
* [ ] Linux release tarball
* [ ] APT package
* [ ] macOS release tarball
* [ ] Homebrew tap
* [ ] Built from source

### What was the issue?

A panic.

### Is the issue reproducible? If so, please provide reproduction instructions.

Yes. I can't share the data and query that trigger the panic. It'd be helpful if I could see which message causes the panic, then I can try to create a standalone repro.

### Please attach any applicable log files.

https://pastebin.com/a25dxrVQ

**Comment by mlazowik** [comment]
I just tried using `materialize/tb:v0.8.0` instead, same panic.

**Comment by philip-stoev** [comment]
@mlazowik would you be able to share the following information, if not sensitive:
- the log from the debezium instance
- the schema for the data or at least some general characteristics about it? Are you using any complex types? The names of the fields are not important so you can obfuscate those if you like.

Thank you!

**Comment by mlazowik** [comment]
Logs: https://gist.github.com/mlazowik/608167e172a057a7c4db13cac3213a83

I didn't see anything interesting in them.

Schema:
```
db=> \d+ e_exedenrow;
                                                        Table "public.e_exedenrow"
        Column         |  Type   | Collation | Nullable |                          Default                          | Storage  | Stats target | Description 
-----------------------+---------+-----------+----------+-----------------------------------------------------------+----------+--------------+-------------
 id_int                | integer |           | not null | nextval('e_exedenrow_id_seq'::regclass)                   | plain    |              | 
 row_number            | integer |           | not null |                                                           | plain    |              | 
 json                  | jsonb   |           | not null |                                                           | extended |              | 
 exed_en_id_int        | integer |           | not null |                                                           | plain    |              | 
 json_typed            | jsonb   |           | not null | '{}'::jsonb                                               | extended |              | 
 ex_id_int             | integer |           |          |                                                           | plain    |              | 
 id                    | uuid    |           | not null |                                                           | plain    |              | 
 en_id                 | uuid    |           | not null |                                                           | plain    |              | 
 ex_id                 | uuid    |           | not null |                                                           | plain    |              | 
 exed_en_id            | uuid    |           | not null |                                                           | plain    |              | 
Indexes:
    "e_exedenrow_pkey" PRIMARY KEY, btree (id_int)
    "e_exedent_exed_en_id_row_nu_0ec386c0_uniq" UNIQUE CONSTRAINT, btree (exed_en_id_int, row_number)
    "e_exedent_exed_en_uuid_row__7bc3f11b_uniq" UNIQUE CONSTRAINT, btree (exed_en_id, row_number)
    "e_exedenrow_uuid_key" UNIQUE CONSTRAINT, btree (id)
    "e_exedenrow_en_uuid_c86109cf" btree (en_id)
    "e_exedenrow_ex_id_587e6b25" btree (ex_id_int)
    "e_exedenrow_ex_uuid_b4760baf" btree (ex_id)
    "e_exedenrow_exed_en_id_97833096" btree (exed_en_id_int)
    "e_exedenrow_exed_en_uuid_70de708f" btree (exed_en_id)
Foreign-key constraints:
    "e_exe_ex_id_587e6b25_fk_excel_exp" FOREIGN KEY (ex_id) REFERENCES e_ex(id) DEFERRABLE INITIALLY DEFERRED
    "e_exe_ex_id_int_d211af3f_fk_excel_exp" FOREIGN KEY (ex_id_int) REFERENCES e_ex(id_int) DEFERRABLE INITIALLY DEFERRED
    "e_exe_exed_en_id_97833096_fk_excel_exp" FOREIGN KEY (exed_en_id) REFERENCES e_exeden(id) DEFERRABLE INITIALLY DEFERRED
    "e_exe_exed_en_id_in_a95aaff1_fk_excel_exp" FOREIGN KEY (exed_en_id_int) REFERENCES e_exeden(id_int) DEFERRABLE INITIALLY DEFERRED
    "e_exedenrow_en_id_2b2bc6c0_fk_api_en_id" FOREIGN KEY (en_id) REFERENCES api_en(id) DEFERRABLE INITIALLY DEFERRED
Publications:
    "dbz_publication"
Replica Identity: FULL
```

The jsons are simple, kv only, no nested objects or even arrays.

**Comment by mlazowik** [comment]
I think I have a specific row this is failing on, will try to narrow down the repro next. Long shot – could this be related to very large numbers in JSON? We have one 20 digit long number as a value under one of the keys.

**Comment by mlazowik** [comment]
Ha, the long shot turns out to be true! Detailed repro instructions: https://gist.github.com/mlazowik/61b0747dd95e3082924bb3d53fb0b890

tl;dr

1. Create a postgresql table with `create table t (j jsonb not null);`
2. Connect debezium
3. `insert into t values ('{"n": 1}');`
4. Connect materialize to that table
5. `insert into t values ('{"n": 1000000000000000000}');` – panic
